src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -499,7 +499,11 @@ private void CheckForHttp11ConnectionInjection()
 
                 RequestQueue<HttpConnection>.QueueItem queueItem = _http11RequestQueue.PeekNextRequestForConnectionAttempt();
 
-                Task.Run(() => AddHttp11ConnectionAsync(queueItem));
+                // Queue the creation of the connection to escape the held lock
+                ThreadPool.QueueUserWorkItem(static state =>
```

Comment:
ThreadPool rather than Task because it's cheaper as you don't need an actual Task as there's no CancellationToken?

Reply:
You don't need a Task because the resulting Task isn't being used.